### PR TITLE
React App Todo-List successfully created following the given requirements

### PR DESCRIPTION
**Pull request by DewDropsDoha**
Fixes #221

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #236

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [x] Infrastructure
- [ ] Other (please specify)

## Requirements
Create a Todo app using React. The app have following requirements:

1. An input field
(a) when the input field is focused, it shows an indicator that it is focused. Show hover effect on the input field. 
(b) Pressing Enter while focusing on input field after writing the task should add the task on the Task list and defocus the input field
(c) Pressing Escape while focusing on input field will clear the text and defocus the input field.

2. Task List
(a) List of tasks designed with pure css 
(b) Hovering over each of the list items should show a delete button and a completed button.
(c) Clicking the delete button should delete the task item and the completed button should keep the item in the list, but it should be strike-through.

3. Use create-react-app 
4. Use React Functional Component and hooks.
5. Rearrange components and folders in an efficient way. 
6. Use React best patterns

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: react/todo-list/src/components/todoList/ToDoList.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
```
Comment (tusher-A) [inline react/todo-list/src/components/todoList/ToDoList.js:1]: This file here seems to be the same file as `TodoList.js` and from index you are exporting. This is totally an unnecessary file. Please remove it.

File: react/todo-list/src/components/todoList/TodoList.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          </div>
+          <div
+            className={
+              task.isCompleted ? "todo-text todo-completed" : "todo-text"
+            }
+          >
+            {toggleButton(task, index)}
+            <button onClick={() => removeTask(index)}>
+              <i class="fa fa-trash-alt" style={{ color: "red" }}></i>
+            </button>
```
Comment (tusher-A) [inline react/todo-list/src/components/todoList/TodoList.js:65]: You should split this into a separate component for better use. Move this to `components/taskItem` folder with proper structure.

**Review by tusher-A — CHANGES_REQUESTED** [review]
Requested some minor changes.

File: react/todo-list/src/components/todoList/TodoList.js (hunk 2 above)
Comment (DewDropsDoha) [reply react/todo-list/src/components/todoList/TodoList.js]: TodoList split-able code moved to new component taskItem folder.
 Commit: 2a59698

File: react/todo-list/src/components/todoList/ToDoList.js (hunk 1 above)
Comment (DewDropsDoha) [reply react/todo-list/src/components/todoList/ToDoList.js:1]: I have renamed the ToDoList.js to TodoList.js. There's no ToDoList.js file in the project. Image referencing no ToDoList.js file in project add to static folder.  Commit 5b4ac9b8

Comment (tusher-A) [inline react/todo-list/src/components/todoList/ToDoList.js:1]: This file here should be deleted. Please verify.

File: react/todo-list/src/components/todoList/components/taskItem/TaskItem.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  useEffect(() => {
+    taskStateSetter(childState);
+  }, [taskStateSetter, childState]);
+
+  const toggleTask = (index) => {
+    const newTasks = [...tasks];
+    newTasks[index].isCompleted = !newTasks[index].isCompleted;
+    setChildState(newTasks);
+  };
```
Comment (tusher-A) [inline react/todo-list/src/components/todoList/components/taskItem/TaskItem.js:14]: You don't need to keep `useEffect` here, as you can see you can directly call the `taskStateSetter` from the `toggleTask` and `removeTask` function. Keeping an extra child state here is unnecessary.

**Review by tusher-A — CHANGES_REQUESTED** [review]
Still some issues remaining. Please verify.

File: react/todo-list/src/components/todoList/ToDoList.js (hunk 1 above)
Comment (DewDropsDoha) [reply react/todo-list/src/components/todoList/ToDoList.js]: Mentioned file deleted. Commit 350ff391

File: react/todo-list/src/components/todoList/components/taskItem/TaskItem.js (hunk 3 above)
Comment (DewDropsDoha) [reply react/todo-list/src/components/todoList/components/taskItem/TaskItem.js]: useEffect removed. taskStateSetter used in toggleTask and removeTask. Commit f25916f1

**Review by tusher-A — APPROVED** [review]
LGTM.

**Pull request merged**